hibernate-core/src/main/java/org/hibernate/query/sqm/internal/AbstractSqmSelectionQuery.java

```diff
@@ -325,10 +325,11 @@ protected TupleMetadata buildTupleMetadata(SqmStatement<?> statement, Class<R> r
 		}
 	}
 
-	private static <R> boolean isTupleMetadataRequired(Class<R> resultType, SqmSelection<?> selection) {
+	private static <R> boolean isTupleMetadataRequired(Class<R> resultType, List<SqmSelection<?>> selections) {
+		final SqmSelection<?> selection = selections.get( 0 );
 		return isHqlTuple( selection )
 			|| !isInstantiableWithoutMetadata( resultType )
-				&& !isSelectionAssignableToResultType( selection, resultType );
+				&& (selections.size() > 1 || !isSelectionAssignableToResultType( selection, resultType ));
```

Comment:
This way the size check is implied by `selection` being `null` if there is more than 1:

```suggestion
		final var selection = selections.size() == 1 ? selections.get( 0 ) : null;
		return isHqlTuple( selection )
			|| !isInstantiableWithoutMetadata( resultType )
				&& !isSelectionAssignableToResultType( selection, resultType );
```